python/ray/data/_internal/execution/operators/union_operator.py

```diff
@@ -37,7 +37,7 @@ def __init__(
         self._input_idx_to_output = 0
 
         self._output_buffer: List[RefBundle] = []
-        self._stats: StatsDict = {}
+        self._stats: StatsDict = {"Union": []}
```

Comment:
initialize stats, mirroring the existing implementation: https://github.com/ray-project/ray/blob/cde9bf790b7cee3d7811b6e5be989bd938c9138e/python/ray/data/dataset.py#L1919-L1922